packages/core/src/audio/AudioSource.ts

```diff
@@ -150,18 +152,34 @@ export class AudioSource extends Component {
    * Play the clip.
    */
   play(): void {
-    if (!this._canPlay()) {
+    if (!this._clip?._getAudioSource()) {
       return;
     }
-    if (this._isPlaying) {
+    if (this._isPlaying || this._pendingPlay) {
       return;
     }
-    const startTime = this._pausedTime > 0 ? this._pausedTime - this._playTime : 0;
-    this._initSourceNode(startTime);
 
-    this._playTime = AudioManager.getContext().currentTime - startTime;
-    this._pausedTime = -1;
-    this._isPlaying = true;
+    if (AudioManager.isAudioContextRunning()) {
+      this._startPlayback();
+    } else {
+      // iOS Safari requires resume() to be called within the same user gesture callback that triggers playback.
+      // Document-level events won't work - must call resume() directly here in play().
```

Comment:
not only safri, but also chrome, see https://developer.chrome.com/blog/autoplay?hl=zh-cn#web_audio